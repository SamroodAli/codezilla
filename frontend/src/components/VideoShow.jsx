import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

const VideoShow = () => {
  const currentVideo = useSelector(({ currentVideo }) => currentVideo);
  if (!currentVideo) {
    return <div>Loading</div>;
  }

  const { description, title } = currentVideo.snippet;
  const videoUrl = `https://www.youtube.com/embed/${currentVideo.id.videoId}`;

  return (
    <div>
      <div className="ui embed">
        <iframe src={videoUrl} frameBorder="0" title={title} />
      </div>
      <div className="ui segment">
        <div className="content">
          <h4 className="ui header">{title}</h4>
          <p>{description}</p>
        </div>
      </div>
    </div>
  );
};

VideoShow.propTypes = {
  video: PropTypes.shape({
    thumbnails: PropTypes.shape({
      medium: PropTypes.shape({
        url: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
    description: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
};
export default VideoShow;
