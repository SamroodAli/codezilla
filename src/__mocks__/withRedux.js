import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
