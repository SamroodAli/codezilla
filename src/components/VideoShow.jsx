import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

const VideoShow = () => {
  const currentVideo = useSelector(({ currentVideo }) => currentVideo);
  if (!currentVideo) {
    return <div>Loading</div>;
  }

  const { thumbnails, description, title } = currentVideo.snippet;

  return (
    <div>
      <img className="ui image" src={thumbnails.medium.url} alt={description} />
      <div className="content">
        <div className="header">{title}</div>
      </div>
    </div>
  );
};

VideoShow.propTypes = {
  video: PropTypes.shape({
    thumbnails: PropTypes.shape({
      medium: PropTypes.shape({
        url: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
    description: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
};
export default VideoShow;
