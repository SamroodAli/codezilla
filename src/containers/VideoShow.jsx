import { useSelector } from 'react-redux';
import useActions from '../hooks/useActions';
import VideoShow from '../components/VideoShow';

const VideoShowContainer = () => {
  const { currentVideo } = useSelector((state) => state);
  const { searchVideo } = useActions();

  return <VideoShow currentVideo={currentVideo} searchVideo={searchVideo} />;
};

export default VideoShowContainer;
