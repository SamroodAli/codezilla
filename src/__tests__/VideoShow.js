import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import withRedux from 'withRedux';
import VideoShow from '../components/VideoShow';

describe('Testing VideoShow with redux', () => {
  it('VideoShow is defined', async () => {
    const component = render(withRedux(<VideoShow />));
    expect(component).toBeDefined();
  });

  it('VideoShow snapshot testing', () => {
    const tree = renderer.create(withRedux(<VideoShow />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
