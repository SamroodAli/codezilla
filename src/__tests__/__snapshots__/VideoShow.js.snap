// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoShow with redux VideoShow snapshot testing 1`] = `
<div>
  Loading
</div>
`;
